server/src/tests/integration/erc20.rs

```diff
@@ -0,0 +1,164 @@
+use {
+    super::*,
+    move_vm_runtime::session::SerializedReturnValues,
+    moved_evm_ext::{extract_evm_result, EVM_NATIVE_ADDRESS},
+    moved_shared::primitives::ToEthAddress,
+};
+
+alloy::sol!(
+    #[sol(rpc)]
+    Erc20,
+    "src/tests/res/ERC20.json"
+);
+mod erc20_factory {
+    alloy::sol!(
+        #[sol(rpc)]
+        OptimismMintableERC20Factory,
+        "src/tests/res/OptimismMintableERC20Factory.json"
+    );
+}
+
+mod bridge {
+    alloy::sol!(
+        #[sol(rpc)]
+        L1StandardBridge,
+        "src/tests/res/L1StandardBridge.json"
+    );
+}
+
+const NAME: &str = "Gold";
+const SYMBOL: &str = "AU";
+
+/// Create a new ERC-20 token on the L1 chain, returning its address.
+/// For convenience, this function also calls `approve` on the new
+/// ERC-20 token allowing the `L1StandardBridgeProxy` to spend the newly
+/// created tokens.
+pub async fn deploy_l1_token(from_wallet: &PrivateKeySigner, rpc_url: &str) -> Result<Address> {
+    let from_address = from_wallet.address();
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .wallet(EthereumWallet::from(from_wallet.to_owned()))
+        .on_http(Url::parse(rpc_url)?);
+
+    let contract = Erc20::deploy(
+        provider,
+        NAME.into(),
+        SYMBOL.into(),
+        from_address,
+        U256::MAX,
+    )
+    .await?;
+
+    let bridge_address = Address::from_str(&get_deployed_address("L1StandardBridgeProxy")?)?;
+    contract
+        .approve(bridge_address, U256::MAX)
+        .send()
+        .await?
+        .watch()
+        .await?;
+
+    Ok(*contract.address())
+}
+
+/// Use the `OptimismMintableERC20Factory` to create a new ERC-20 token on the L2.
+/// This token is used for bridging the L1 token with the given address to the L2.
+pub async fn deploy_l2_token(
+    from_wallet: &PrivateKeySigner,
+    l1_address: Address,
+    rpc_url: &str,
+) -> Result<Address> {
+    let factory_address = alloy::primitives::address!("4200000000000000000000000000000000000012");
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .wallet(EthereumWallet::from(from_wallet.to_owned()))
+        .on_http(Url::parse(rpc_url)?);
+
+    let contract = erc20_factory::OptimismMintableERC20Factory::new(factory_address, provider);
+    let receipt = contract
+        .createOptimismMintableERC20(l1_address, NAME.into(), SYMBOL.into())
+        .send()
+        .await
+        .unwrap()
```

Comment:
Maybe just `?` all the unwraps in this func and the balance one as we're returning `anyhow` result anyways?

Reply:
I was unwrapping just for debugging purposes since then the specific line number shows up in the error message, whereas `?` will propagate the error back to the test runner. We could mitigate this issue by using the `context` machinery that `anyhow` offers, but for now I'll just put in the question marks.

Done in https://github.com/MovedNetwork/op-move/pull/307/commits/5ece865158ea1c6db900db6338c90bc90cfd9c7b